## Patch originally from Fedora - Revision 1.4 of:
http://cvs.fedoraproject.org/viewvc/rpms/vnc/devel/vnc-xorg.patch
## Updated to apply to Xorg 1.5.99.901

diff -urp -x '*~' -x '*.orig'  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/configure.ac  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/configure.ac
---  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/configure.ac	2009-01-23 17:12:07.905592000 -0800
+++  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/configure.ac	2009-01-23 17:16:49.289083000 -0800
@@ -29,7 +29,6 @@ AC_PREREQ(2.57)
 AC_INIT([xorg-server], 1.5.99.901, [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg], xorg-server)
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([dist-bzip2 foreign])
-AM_MAINTAINER_MODE
 
 dnl this gets generated by autoheader, and thus contains all the defines.  we
 dnl don't ever actually use it, internally.
@@ -1272,6 +1271,9 @@ if test "x$XVFB" = xyes; then
 	AC_SUBST([XVFB_SYS_LIBS])
 fi
 
+dnl Xvnc DDX
+AC_SUBST([XVNC_CPPFLAGS], ["-DHAVE_DIX_CONFIG_H $XEXT_INC $FB_INC $MI_INC $RENDER_INC"])
+AC_SUBST([XVNC_LIBS], ["$FB_LIB $FIXES_LIB $XEXT_LIB $CONFIG_LIB $DBE_LIB $RECORD_LIB $GLX_LIBS $RANDR_LIB $RENDER_LIB $DAMAGE_LIB $MIEXT_DAMAGE_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $COMPOSITE_LIB"])
 
 dnl Xnest DDX
 
@@ -1926,6 +1928,7 @@ hw/dmx/input/Makefile
 hw/dmx/glxProxy/Makefile
 hw/dmx/Makefile
 hw/vfb/Makefile
+hw/vnc/Makefile
 hw/xnest/Makefile
 hw/xwin/Makefile
 hw/xquartz/Makefile
diff -urp -x '*~' -x '*.orig'  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/hw/Makefile.am  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/hw/Makefile.am
---  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/hw/Makefile.am	2008-10-12 16:18:43.000000000 -0700
+++  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/hw/Makefile.am	2009-01-23 17:16:49.414933000 -0800
@@ -33,7 +33,8 @@ SUBDIRS =			\
 	$(XNEST_SUBDIRS)	\
 	$(DMX_SUBDIRS)		\
 	$(KDRIVE_SUBDIRS)	\
-	$(XQUARTZ_SUBDIRS)
+	$(XQUARTZ_SUBDIRS)	\
+	vnc
 
 DIST_SUBDIRS = dmx xfree86 vfb xnest xwin xquartz kdrive
 
diff -urp -x '*~' -x '*.orig'  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/mi/miinitext.c  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/mi/miinitext.c
---  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/mi/miinitext.c	2009-01-23 17:12:07.969104000 -0800
+++  vnc-4_1_3-unixsrc/unix/xorg-x11-server-source/mi/miinitext.c	2009-01-23 17:16:49.493023000 -0800
@@ -219,6 +219,7 @@ extern void XInputExtensionInit(INITARGS
 extern void XTestExtensionInit(INITARGS);
 #endif
 extern void BigReqExtensionInit(INITARGS);
+extern void vncExtensionInit(INITARGS);
 #ifdef SCREENSAVER
 extern void ScreenSaverExtensionInit (INITARGS);
 #endif
@@ -432,6 +433,7 @@ InitExtensions(int argc, char *argv[])
     if (!noTestExtensions) XTestExtensionInit();
 #endif
     BigReqExtensionInit();
+    vncExtensionInit();
 #if defined(SCREENSAVER)
     if (!noScreenSaverExtension) ScreenSaverExtensionInit ();
 #endif
